File: packages/ts-sdk/test/adapters/storage/artifacts.test.ts
import { describe, expect, it } from 'vitest';
import { ArtifactsAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('ArtifactsAdapter', () => {
  it('constructs artifact endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new ArtifactsAdapter({ transport });
    await adapter.export('repo', {});
    await adapter.list('repo');
    await adapter.get('repo', 'artifact');
    await adapter.delete('repo', 'artifact');
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/artifacts/export',
      'GET /api/v1/repos/repo/artifacts',
      'GET /api/v1/repos/repo/artifacts/artifact',
      'DELETE /api/v1/repos/repo/artifacts/artifact'
    ]);
  });
});

File: packages/elixir-sdk/test/adapters/storage/artifacts_test.exs
defmodule TreeDxSdk.ArtifactsAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Artifacts.export(client, "repo/a", %{})
    TreeDxSdk.Artifacts.list(client, "repo/a")
    TreeDxSdk.Artifacts.get(client, "repo/a", "art/a")
    TreeDxSdk.Artifacts.delete(client, "repo/a", "art/a")

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :delete and &1.path == "/api/v1/repos/repo%2Fa/artifacts/art%2Fa")
           )
  end
end
